### PR TITLE
Quatity update

### PR DESCRIPTION
**Pull request by Flyernav**
Delete functionality works, and added ( pk: res.data.pk )on .then to addOnProduct Function in GroceryListDetail

### CODE REVIEW

**Review by Marsigli — APPROVED** [review]

File: grocerease/src/components/GroceryListItem.js [hunk 1]
```diff
@@ -1,17 +1,52 @@
 import { useState } from 'react'
+import axios from 'axios';
 
-const GroceryListItem = ({item}) => {
+
+const GroceryListItem = ({item, token}) => {
     console.log(item)
-    const [itemCount, setItemCount] = useState(item.item_quantity);
+    const [itemData, setItemData] = useState(item);
+    const [itemCount,setItemCount] = useState();
```
Comment (amygori) [inline grocerease/src/components/GroceryListItem.js:8]: These two lines are going to cause major bugs. You can't take data from props and put them into state.

File: grocerease/src/components/GroceryListItem.js [hunk 2]
```diff
@@ -1,17 +1,52 @@
...
+    const deleteItem = (item) => {
+        axios.delete(`https://grocerease.herokuapp.com/grocerease/delete_item/${item.pk}/`,
+        {
+            headers: {
+                'Content-Type': 'application/json',
+                Authorization: `token ${token}`
+            },
+        })
+        .then (res => {
+            setItemData(item.filter(itemData => (item !== item.pk)))
```
Comment (amygori) [inline grocerease/src/components/GroceryListItem.js:19]: What is supposed to be happening here?

File: grocerease/src/components/GroceryListItem.js [hunk 3]
```diff
@@ -1,17 +1,52 @@
...
+                    value={itemCount}
+                    onChange={(event) => {
+                        setItemCount(event.target.value);
+                        console.log({itemCount})
+                        axios.patch(`https://grocerease.herokuapp.com/grocerease/item_detail/${item.pk}/`,
+                        console.log(event),
+                        {
+                         quantity: item.item_quantity,
+                        },console.log(item.item_quantity))}}>    
+                </input> */}
```
Comment (amygori) [inline grocerease/src/components/GroceryListItem.js:48]: This needs to be deleted. We don't want to commit commented out code.

**Review by amygori — CHANGES_REQUESTED** [review]
@Flyernav Did you check to make sure that this code works with the existing code (making sure to merge in the latest from main?). There are some changes you will need to make, or else this code will introduce bugs to your application.

**Comment by amygori** [comment]
@Flyernav This code should be on a different branch from the quantity update work, since we decided not to continue developing that yesterday. At this point I'd suggest at least renaming the pull request to match the feature that this actually is.

**Pull request closed without merging**